src/Raven.Server/Utils/ThreadIoStatsReader.cs

```diff
@@ -0,0 +1,278 @@
+            public double KbPerSecLast;
+            public long OpsTotal;
+            public double KbTotal;
+
+            // Split metrics when available
+            public double? ReadOpsPerSecLast;
+            public double? WriteOpsPerSecLast;
+            public double? ReadKbPerSecLast;
+            public double? WriteKbPerSecLast;
+            public long? ReadOpsTotal;
+            public long? WriteOpsTotal;
+            public double? ReadKbTotal;
+            public double? WriteKbTotal;
+        }
+
+        private ThreadIoStatsReader()
+        {
+            if (PlatformDetails.RunningOnLinux == false)
+                return;
+
+            try
+            {
+                _timer = new Timer(_ =>
+                {
+                    if (Interlocked.Exchange(ref _collecting, 1) != 0)
+                        return;
+                    try
+                    {
+                        CollectOnce();
+                    }
+                    catch (Exception e)
+                    {
+                        if (Logger.IsInfoEnabled)
+                            Logger.Info("Error in thread IO stats collector.", e);
+                    }
+                    finally
+                    {
+                        Volatile.Write(ref _collecting, 0);
+                    }
+                }, null, TimeSpan.Zero, TimeSpan.FromSeconds(1));
+            }
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info("Failed to start timer for thread IO stats collector. Thread IO stats disabled.", e);
+            }
+        }
+
+        private sealed class PrevSample
+        {
+            public long Syscr;
+            public long Syscw;
+            public long ReadBytes;
+            public long WriteBytes;
+            public DateTime Last;
+        }
+
+
+        private static ThreadIoStatsReader _instance;
+        private static readonly object InstanceLock = new object();
+
+        public static ThreadIoStatsReader Instance
+        {
+            get
+            {
+                if (_instance != null)
+                    return _instance;
+                lock (InstanceLock)
+                {
+                    return _instance ??= new ThreadIoStatsReader();
+                }
+            }
+        }
+
+        public static void DisposeInstance()
+        {
+            lock (InstanceLock)
+            {
+                try
+                {
+                    _instance?.Dispose();
+                }
+                catch
+                {
+                    // ignore
+                }
+                finally
+                {
+                    _instance = null;
+                }
+            }
+        }
+
+        public bool TryGet(int tid, out Stats stats) => _stats.TryGetValue(tid, out stats);
+
+        public void Dispose()
+        {
+            try
+            {
+                _timer?.Dispose();
```

Comment:
Shouldn't you clear `_stats` and `_prevSamples`? Can we `null` the `_timer` instance?

Also please note that the timer's `Dispose()` method doesn't guarantee that the callback isn't currently executing. I don't think it can affect us but please consider that.